### PR TITLE
Support externalId auth

### PR DESCRIPTION
**Pull request by liujoshua**
Depends on this to build: https://github.com/Sage-Bionetworks/BridgeJavaSDK/pull/509

### CODE REVIEW

**Comment by liujoshua** [comment]
Depends on:
https://github.com/Sage-Bionetworks/BridgeAndroidSDK/pull/167
https://github.com/Sage-Bionetworks/BridgeJavaSDK/pull/509

File: android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/AuthenticationManager.java [hunk 1]
```diff
@@ -269,10 +286,9 @@ public Completable signUp(@NonNull final SignUp signUp) {
      * Signs up just using a phone number. No username/password will be stored at this time.
      * <p>
      *
-     * @param phone  participant's phone number.
-     *               To prevent accidental improper settings, study will be set to the studyId
-     *               provided in the Bridge, consent will be set to false, and account status
-     *               will be cleared.
+     * @param phone
+     *         participant's phone number. To prevent accidental improper settings, study will be set to the studyId
+     *         provided in the Bridge, consent will be set to false, and account status will be cleared.
```
Comment (dephillipsmichael) [inline android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/AuthenticationManager.java:301]: As we move towards iOS parity, I'd suggest deprecating old method names.  For instance, on iOS...

signUp = signUpStudyParticipant
signIn = signInWithEmail, signInWithExternalID, signInWithCredential, etc.

File: android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/AuthenticationManager.java [hunk 2]
```diff
@@ -441,16 +448,32 @@ public Completable requestPhoneSignIn(@NonNull String regionCode, @NonNull Strin
                 .compose(signInHelper(signIn));
     }
 
+    @NonNull
+    public Single<UserSessionInfo> signInViaExternalId(@NonNull String externalId, @NonNull String password) {
```
Comment (dephillipsmichael) [inline android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/AuthenticationManager.java]: signInWithExternalId

**Comment by dephillipsmichael** [comment]
We have some dependencies on having a valid email in BridgeDataProvider as well that need attention. 

```
public boolean isSignedUp() {
        logger.debug("Called isSignedUp");
        return authenticationManager.getEmail() != null;
}
```

```
    public boolean isSignedIn() {
        logger.debug("Called isSignedIn");
        return authenticationManager.getEmail() != null &&
                authenticationManager.getUserSessionInfo() != null;
    }
```

```
    @NonNull
    public Single<UserSessionInfo> setUserSharingScope(@Nullable SharingScope scope) {
        logger.debug("Called setUserSharingScope with: " + scope);

        return bridgeManagerProvider.getParticipantManager()
                .updateParticipantRecord((StudyParticipant) new StudyParticipant()
                        .email(authenticationManager.getEmail())
                        .sharingScope(scope))
                .doOnSuccess(session -> bridgeManagerProvider.getAccountDao()
                        .setDataGroups(session.getDataGroups()));
    }

```

**Pull request merged**